### DISCUSSION TITLE
Would it be possible to export `matchMutation` from the library?

### DISCUSSION THREAD

**Discussion by luke-layerhealth** [TanStack/query#6802] [Ideas]
Would it be possible to export `matchMutation` from the library similar to how the library exports `matchQuery`?

https://github.com/TanStack/query/blob/c8442f8256aa37b479a45037a1681066f001496b/packages/query-core/src/utils.ts#L139

The use case is setting up a `MutationCache` with a default `onSettled` handler. I want to check if the mutation key matches a filter then fire some default query invalidation. While I can role my own mutation key matching it seems cleaner to use `matchMutation`.

**Comment by TkDodo** [comment]
oh, we export `matchQuery` but not `matchMutation`. sure, PRs welcome

**Reply by luke-layerhealth** [reply]
Submitted a PR #6851
